.github/workflows/pr_checks.yml

```diff
@@ -0,0 +1,35 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  pull_request:
+    branches: [ master ]
```

Comment:
root branch is `main`

Reply:
also add this action for lint checking in the pipeline: https://github.com/marketplace/actions/eslint-plus

Reply:
done